### PR TITLE
#195 Fix invocation service to work with OCI helm chart

### PR DESCRIPTION
**Pull request by carter-cundiff**

### CODE REVIEW

File: extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml [hunk 1]
```diff
@@ -24,58 +24,9 @@
...
-                                        </image>
-                                    </images>
-                                </configuration>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml:55]: I: Removed as this build argument is not used in this dockerfile

File: extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml [hunk 2]
```diff
@@ -24,58 +24,9 @@
...
-                        <artifactId>docker-maven-plugin</artifactId>
-                        <executions>
-                            <execution>
-                                <id>default-push</id>
-                                <configuration>
-                                    <images>
-                                        <image>
-                                            <build>
-                                                <args>
-                                                    <AISSEMBLE_HELM_REPO>${aissemble.helm.repo}</AISSEMBLE_HELM_REPO>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml:43]: I: Removed as this build argument is not used in this dockerfile

File: extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml [hunk 3]
```diff
@@ -24,58 +24,9 @@
...
-                <artifactId>docker-maven-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>default-build</id>
-                        <configuration>
-                            <images>
-                                <image>
-                                    <build>
-                                        <args>
-                                            <AISSEMBLE_HELM_REPO>${aissemble.helm.repo}</AISSEMBLE_HELM_REPO>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml:70]: I: See above

File: extensions/extensions-docker/aissemble-pipeline-invocation/src/main/resources/docker/Dockerfile [hunk 4]
```diff
@@ -6,8 +6,6 @@ FROM ${DOCKER_BASELINE_REPO_ID}boozallen/aissemble-quarkus:${VERSION_AISSEMBLE}
 LABEL org.opencontainers.image.source="https://github.com/boozallen/aissemble"
 
 USER root
-ARG AISSEMBLE_HELM_REPO=oci://ghcr.io/boozallen
-ARG VERSION_KUBERNETES=1.28
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-pipeline-invocation/src/main/resources/docker/Dockerfile:10]: I: Removed unused `ARG`s

File: extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml (hunk 2 above)
Comment (Cho-William) [reply extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml:43]: Just curious, why was the helm repo url being given to docker in the first place? I know you said it was unused, but I wonder why this was simply just a mistake from before?

File: extensions/extensions-pipeline-invocation-service/src/main/java/com/boozallen/aissemble/pipeline/invocation/service/util/exec/HelmCommandExecutor.java [hunk 5]
```diff
@@ -102,8 +104,7 @@ public List<String> createBaseHelmInstallArgs(String appName) {
 
         args.add("install");
         args.add(appName);
-        args.add("--repo");
-        args.add(aissembleHelmRepoUrl);
+        args.add(aissembleHelmRepoProtocol + "://" + aissembleHelmRepoUrl + "/aissemble-spark-application-chart");
```
Comment (Cho-William) [inline extensions/extensions-pipeline-invocation-service/src/main/java/com/boozallen/aissemble/pipeline/invocation/service/util/exec/HelmCommandExecutor.java:107]: Q: so it seems like this is the main fix? It was just missing the protocol?

**Review by Cho-William — APPROVED** [review]

File: extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml (hunk 2 above)
Comment (carter-cundiff) [reply extensions/extensions-docker/aissemble-pipeline-invocation/pom.xml:43]: Seems like it was mistakenly added in the initial build out of the invocation service and was never caught.

File: extensions/extensions-pipeline-invocation-service/src/main/java/com/boozallen/aissemble/pipeline/invocation/service/util/exec/HelmCommandExecutor.java (hunk 5 above)
Comment (carter-cundiff) [reply extensions/extensions-pipeline-invocation-service/src/main/java/com/boozallen/aissemble/pipeline/invocation/service/util/exec/HelmCommandExecutor.java:107]: Yep, this [SO](https://stackoverflowteams.com/c/boozallensolutioncenter/questions/562/563#563) sums it up well

**Pull request merged**